src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -680,22 +651,22 @@ public ReadOnlyTensorSpan<T> Slice(params scoped ReadOnlySpan<NRange> ranges)
             if (ranges.Length != Lengths.Length)
                 throw new ArgumentOutOfRangeException(nameof(ranges), "Number of dimensions to slice does not equal the number of dimensions in the span");
 
-            scoped Span<nint> shape;
+            scoped Span<nint> lengths;
             scoped Span<nint> offsets;
-            if (Rank > 6)
+            if (Rank > TensorSpan.MaxRankForStackAlloc)
```

Comment:
nit: The temporary buffer used here has nothing to do with the limits set by `MaxRankForStackAlloc`.

If we're within `MaxRankForStackAlloc` then we're going to be copying the ranks from the stackalloc here to the inline fields of the `TensorShape`; otherwise we're going to be allocating the internal `nint[] _metadata` array and copying it into that.

As such, this stackalloc only needs to consider the normal stackalloc limits and can use `128` as the boundary at which to stackalloc or not.

We could likewise avoid 2x allocations by simply having a single `scoped Span<int> metadata` that is `Rank * 2` in length and then slicing `lengths` and `offsets` out of that, which should make things simpler and cheaper in the fallback path as well.